cognite/pygen/_query/interface.py

```diff
@@ -313,22 +376,38 @@ def _execute_aggregation(
             raise ValueError("No aggregation found")
 
     @staticmethod
-    def _metric_aggregation_to_dict(aggregation: list[dm.aggregations.AggregatedNumberedValue]) -> dict[str, Any]:
-        values_by_aggregations: dict[str, dict[str, Any]] = defaultdict(dict)
+    def _update_metric_results(
+        aggregation: list[dm.aggregations.AggregatedNumberedValue], existing: dict[str, dict[str, float | int | None]]
+    ) -> None:
         for item in aggregation:
-            values_by_aggregations[item._aggregate][item.property] = item.value
-        return dict(values_by_aggregations)
+            if item._aggregate not in existing:
+                existing[item._aggregate] = {}
+            existing_value = existing[item._aggregate].get(item.property, None)
+            if existing_value is None:
+                existing[item._aggregate][item.property] = item.value
+            elif item.value is not None:
+                if isinstance(item, dm.aggregations.SumValue | dm.aggregations.CountValue):
+                    new_value = existing_value + item.value
+                elif isinstance(item, dm.aggregations.MaxValue):
+                    new_value = max(existing_value, item.value)
+                elif isinstance(item, dm.aggregations.MinValue):
+                    new_value = min(existing_value, item.value)
+                elif isinstance(item, dm.aggregations.AvgValue):
+                    # We will never call this more than twice, so we can just average the two values.
+                    new_value = (existing_value + item.value) / 2
```

Comment:
I am not totally sure about the end result of this function, but this seems a bit arbitrary. This metrics are updated where, in the end?

Reply:
Added a docstring to explain why this is here.